### PR TITLE
ppj09

### PR DESCRIPTION
**Pull request by valeria1604**

### CODE REVIEW

File: src/com/company/ppj09/Tasks.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    private static void task02() {
+        for (int i = 1; i <= 10; i++) {
+            System.out.println(1 / Math.pow(2, i));
+        }
+    }
+
+    private static void task03() {
+        for (int i = 1; i <= 15; i++) {
```
Comment (maxline) [inline src/com/company/ppj09/Tasks.java]:
```suggestion
        for (byte i = 1; i <= 15; i++) {
```

File: src/com/company/ppj09/Tasks.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            convertByteToHexAndSaveAsChar((byte) i);
+        }
+    }
+
+    private static void convertByteToHexAndSaveAsChar(byte a) {
+        String hexadecimalValue = String.format("%02x", a);
```
Comment (maxline) [inline src/com/company/ppj09/Tasks.java]: this code doesn't work, use `char ch = Integer.toHexString(a).charAt(0);` instead

File: src/com/company/ppj09/Tasks.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                } else {
+                    System.out.print("x");
+                }
+            }
+            if (rows < widthAndHeight / 2 + 1) {
+                x++;
+            } else {
+                x--;
+            }
+            System.out.println();
```
Comment (maxline) [inline src/com/company/ppj09/Tasks.java:53]: try to simplify solution without using additional variable `x`. you can divide algorithm into two parts, first part will print first half of the lines , and second part will print the second half of the lines.

**Review by maxline — CHANGES_REQUESTED** [review]
good job! let's improve the solution.

File: src/com/company/ppj09/Tasks.java (hunk 1 above)
Comment (valeria1604) [reply src/com/company/ppj09/Tasks.java]: fixed

File: src/com/company/ppj09/Tasks.java (hunk 2 above)
Comment (valeria1604) [reply src/com/company/ppj09/Tasks.java]: fixed

File: src/com/company/ppj09/Tasks.java (hunk 3 above)
Comment (valeria1604) [reply src/com/company/ppj09/Tasks.java:53]: fixed

**Review by maxline — APPROVED** [review]
Approved

**Pull request merged**